Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/components/ViewSettingsController.java

```diff
@@ -73,28 +74,38 @@
             "org:resource",
             "org:role"
     );
+    
+    private String[] measures = {
+            "case_count",
+            "case_relative",
+            "total",
+            "mean",
+            "median",
+            "max",
+            "min"
+    };
 
     private Map<String, MeasureAggregation> measureAggMap = new HashMap<String, MeasureAggregation>() {
```

Comment:
If this map is immutable, would it make sense for this code to be moved to MeasureAggregation instead?

Reply:
The purpose of this map is associating UI keys with types of MeasureAggregation used in the weight calculation. For example, the UI key can be a list box item key = "total" which is mapped to MeasureAggregation.TOTAL. It is better to keep a data class design independent from UI keys. In one UI, the key value can be "total" and so there's a mapping from "total" to MeasureAggregation.TOTAL for that specific UI controller (here is ViewSettingsController); in other UI, the key value can be "sum" and so there's a mapping from "sum" to MeasureAggregation.TOTAL. It doesn't matter immutable or not in this class design decision, but the separation between UI and data object is the key decision factor to keep them decoupled and easy to change later.